### ISSUE TITLE
My Electric app not updating

### ISSUE THREAD

**Issue by AndyRap** [emoncms/app#13]
Hi,

I have a problem whereby the app does not update until I refresh the screen.

Looking at the developer console I see the following error -

![app1](https://cloud.githubusercontent.com/assets/3307706/14061465/a2118bcc-f378-11e5-9500-592c5e43c7ef.PNG)

which seems to be tied to the data refresh period -

![app2](https://cloud.githubusercontent.com/assets/3307706/14061477/ca2c7e32-f378-11e5-967c-7b8697bda7d9.PNG)

Although the developer screen shots above are from Microsoft Edge, I have just tried it with Google Chrome and the error is the same.

Any advice on how to correct the issue would be welcome.

Thanks in advance
Andy Hodges

**Comment by AndyRap** [comment]
Fixed,

I don't know JavaScript, but......

Removing the [0] from line 516 -

var next = valid[valid.length-1][0] + (interval*1000);

so as to leave the line as -

var next = valid[valid.length-1] + (interval*1000);

Has fixed the issue.

Kind Regards
Andy Hodges

**Comment by glynhudson** [comment]
Hi Andy, Thanks for reporting issue. I have been unable to re-create. 

What version of Emoncms are you running? 
What browser are you using? 
If using an emonBase/emonPi have you updated to latest version using emonPi update? 

Please could you provide some more info regarding the issue:

Does MyElectric no update at all?
 Or does it stop updating after a period of time?
When does the error in the console appear?

**Comment by AndyRap** [comment]
Hi Glyn,

What version of Emoncms are you running? 
Emoncms 9.4

What browser are you using? 
Edge and Chrome, the issue existed in both.

If using an emonBase/emonPi have you updated to latest version using emonPi update? 
N/A, vanilla Pi build as per method below.

Does MyElectric no update at all?
MyElectric would update if I clicked the refresh button or if I chose a different time view for the bar graph but not if I just let it sit there.

Or does it stop updating after a period of time?
As soon as the page is loaded the value would not change without interaction as per the above answer.

When does the error in the console appear? 
The console error appeared as soon as the MyElectric page was loaded.

_METHOD_
I am running on a Raspberry Pi 2.  All code was fresh from github and was installed on a fresh and updated Raspbian Jessie.

1.  Followed EmonCMS instructions for installing to a Pi, skipped the EmonHub installation.
2. Created new user, inputs, feeds etc.
3.  Installed the dashboard module using cd /var/www/emoncms/Modules && git clone https://github.com/emoncms/dashboard.git
4. Ran database update twice, as required.
5. Installed the app module using cd /var/www/emoncms/Modules && git clone https://github.com/emoncms/app.git
6. Ran the database update once, as required.
7. Set the feeds (Power and kWh) and the price per unit and saved.

That is what I did to get to the point that generated the error in the Microsoft Edge developer console and, as I know MS stuff can be fussy I also checked but didn't screenshot, the same error in Google Chrome developer console.

The error also existed on the WAMP/LAMP Windows 10 install I have setup to do some fiddling / dev work.  I will change the file back and see if I can reproduce the issue.

Right, just done that and it updates :8ball: 

The error only exists if the feeds that have been set have less than one days data in them, that is, the feed has not yet gone past midnight.

I have just created a temp power and kWh feed and assigned them to the MyElectric app and the error has returned.

Kind Regards
Andy Hodges



Kind Regards
Andy Hodges
